plugin.js

```diff
@@ -0,0 +1,139 @@
+const path = require("path");
+const fs = require("fs");
+const shell = require("shelljs");
+const { JSDOM } = require("jsdom");
+
+const Config = require("./Config.js");
+const { PluginScope } = require("./Enums.js");
+
+function setCurr(component, targetPath, subTaskFlag=false) {
+	let obj = {...component}, isCurrentItem = false;
+
+	if(obj.unit_type === "aim")
+	{
+		subTaskFlag = true;
+		isCurrentItem = true;
+	}
+
+	if(!subTaskFlag)
+	{
+		obj = component.menuItemInfo(targetPath);
+	}
+
+	if(obj.unit_type === "lu")
+	{
+		obj.units = [...obj.units.map((subComponent) => setCurr(subComponent, targetPath, true))];
+	}
+
+	return { ...obj, isCurrentItem: isCurrentItem };
+};
+
+class Plugin {
+
+  static getConfigFileName(options_env) {
+    const env = options_env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    return pluginConfigFile;
+  }
+
+  static processExpScopePlugins(exp_info, hb, lab_data, options) {
+    const env = options.env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    const pluginConfig = require(pluginConfigFile);
+
+    const expScopePlugins = pluginConfig.filter(
+      (p) => p.scope === PluginScope.EXPERIMENT
+    );
+
+    let plugins = [];
+    shell.exec('mkdir plugins');
+    expScopePlugins.forEach((plugin) => {
+	    try {
+		    if(!fs.existsSync(path.join('plugins', plugin.id)))
+		    {
+			    shell.cd('plugins');
+			    shell.exec('git clone ' + plugin.repo);
+			    shell.exec(`rm -rf ${path.join(plugin.id, '.git')}`);
+			    shell.cd('..');
+		    }
+		    shell.exec('cp -ur \'' + path.resolve('./plugins') + '\' \'' + Config.build_path(exp_info.src) + '\'');
+
+		    const pluginPath = path.resolve('plugins', plugin.id);
+		    const page_template = fs.readFileSync(path.resolve(pluginPath, plugin.template));
+
+		    let assets_path = path.relative(
+			    path.dirname(path.join(Config.build_path(exp_info.src), plugin.target)),
+			    Config.build_path(exp_info.src)
+		    );
+		    assets_path = assets_path ? assets_path : ".";
+		    const cssModule = plugin.cssModule || "css/main.css";
+		    const jsModule = plugin.jsModule || "js/main.js";
```

Comment:
Please create an enum in the Enum.js named as Plugin and add these entries as following:
const Plugin = {
  Default: {
    JS-MODULE: 'js/main.js'',
    CSS-MODULE: 'css/main.css',
  }
};

Use Plugin.Default.JS-MODULE and Plugin.Default.CSS-MODULE here. Later we will merge the Scope enum into this too leaving one enum for all Plugin configuration.